### ISSUE TITLE
Budgie panel - touchscreen long press to right click on Budgie panel

### ISSUE THREAD

**Issue by JacekJagosz** [solus-project/budgie-desktop#1945]
### Current Budgie version
10.5.1 on Solus 4.1

### Use-case: nice to have for touchscreen laptops, must to have for tablets
Currently you can't access some system tray options, but most crucially you can't switch between a few windows of the same app just using the icon task list and touchscreen. E.g. if you minimize a window on Nautilus and have 2 other instances opened, you won't be able to access without closing other windows or using a mouse.

### Proposal:
Treat touchscreen longpress as a right click on budgie panel, as all Gnome apps and many GTK apps already do. Rest of the Budgie experience works really well on touchscreen.

**Comment by JoshStrobl** [comment]
This is a GTK bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/2784

**Comment by JoshStrobl** [comment]
Ignore me, accidentally posted on your issue and closing it instead of another :man_facepalming: 

But now that I'm here, yes I agree we should implement the GestureLongPress functionality across the board. Might start with IconTasklist, see if it works for you (since I don't have a touch screen) and work from there, if you'd be open to testing it.

**Comment by JacekJagosz** [comment]
I'm eager to test it, I'm using Solus unstable (I'm "Jacek" on dev.getsol.us)

**Comment by mgrouch** [comment]
You can try this evdev-rce (bareboat-necessities fork). It will add right-click emulation via long touch everywhere (in all apps)
with no changes to them.

https://github.com/bareboat-necessities/evdev-right-click-emulation

Binaries
https://launchpad.net/~bbn-projects/+archive/ubuntu/bbn-ppa/

Some docs:

https://github.com/bareboat-necessities/my-bareboat/blob/master/right-click-emu/right-click-emu.sh

Thanks

**Comment by JoshStrobl** [comment]
Or I could implement proper Gtk Gesture support...

**Comment by mgrouch** [comment]
In all applications you have for your desktop???

**Comment by JoshStrobl** [comment]
No. The issue in question is related to Budgie specifically, not "all applications". Please don't hijack the thread to discuss options not relevant to the task, which is Budgie support. Individual applications should be responsible for how and where they implement touch support (if at all).
